util/awsservice/constant.go

```diff
@@ -38,16 +39,41 @@ var (
 )
 
 var (
-	ctx                  = context.Background()
-	awsCfg, _            = config.LoadDefaultConfig(ctx)
-	Ec2Client            = ec2.NewFromConfig(awsCfg)
-	EcsClient            = ecs.NewFromConfig(awsCfg)
-	SsmClient            = ssm.NewFromConfig(awsCfg)
-	ImdsClient           = imds.NewFromConfig(awsCfg)
-	CwmClient            = cloudwatch.NewFromConfig(awsCfg)
-	CwlClient            = cloudwatchlogs.NewFromConfig(awsCfg)
-	DynamodbClient       = dynamodb.NewFromConfig(awsCfg)
-	S3Client             = s3.NewFromConfig(awsCfg)
-	CloudformationClient = cloudformation.NewFromConfig(awsCfg)
-	XrayClient           = xray.NewFromConfig(awsCfg)
+	ctx context.Context
+
+	// AWS Clients
+	Ec2Client            *ec2.Client
+	EcsClient            *ecs.Client
+	SsmClient            *ssm.Client
+	ImdsClient           *imds.Client
+	CwmClient            *cloudwatch.Client
+	CwlClient            *cloudwatchlogs.Client
+	DynamodbClient       *dynamodb.Client
+	S3Client             *s3.Client
+	CloudformationClient *cloudformation.Client
+	XrayClient           *xray.Client
 )
+
+func init() {
```

Comment:
why do we want to initialise separately  instead as variables like before

Reply:
To assign region to awsConfig otherwise getMetricData method fail due to awsConfig not having a region.